pono.cpp

```diff
@@ -133,123 +133,52 @@ const option::Descriptor usage[] = {
 /*********************************** end Option Handling setup
  * ***************************************/
 
-ProverResult check_prop(Engine engine,
-                        unsigned int bound,
+ProverResult check_prop(PonoOptions pono_options,
                         Property & p,
                         SmtSolver & s,
                         SmtSolver & second_solver,
-                        std::vector<UnorderedTermMap> & cex,
-                        const bool no_witness)
+                        std::vector<UnorderedTermMap> & cex)
 {
   logger.log(1, "Solving property: {}", p.prop());
 
   logger.log(3, "INIT:\n{}", p.transition_system().init());
   logger.log(3, "TRANS:\n{}", p.transition_system().trans());
 
   std::shared_ptr<Prover> prover;
-  if (engine == BMC) {
-    prover = std::make_shared<Bmc>(p, s);
-  } else if (engine == BMC_SP) {
-    prover = std::make_shared<BmcSimplePath>(p, s);
-  } else if (engine == KIND) {
-    prover = std::make_shared<KInduction>(p, s);
-  } else if (engine == INTERP) {
+  if (pono_options.engine_ == BMC) {
+    prover = std::make_shared<Bmc>(pono_options, p, s);
+  } else if (pono_options.engine_ == BMC_SP) {
+    prover = std::make_shared<BmcSimplePath>(pono_options, p, s);
+  } else if (pono_options.engine_ == KIND) {
+    prover = std::make_shared<KInduction>(pono_options, p, s);
+  } else if (pono_options.engine_ == INTERP) {
     assert(second_solver != NULL);
-    prover = std::make_shared<InterpolantMC>(p, s, second_solver);
+    prover = std::make_shared<InterpolantMC>(pono_options, p, s, second_solver);
   } else {
     throw PonoException("Unimplemented engine.");
   }
 
-  ProverResult r = prover->check_until(bound);
-  if (r == FALSE && !no_witness) {
+  ProverResult r = prover->check_until(pono_options.bound_);
+  if (r == FALSE && !pono_options.no_witness_) {
     prover->witness(cex);
   }
   return r;
 }
 
 int main(int argc, char ** argv)
 {
-  argc -= (argc > 0);
-  argv += (argc > 0);  // skip program name argv[0] if present
-  option::Stats stats(usage, argc, argv);
-  std::vector<option::Option> options(stats.options_max);
-  std::vector<option::Option> buffer(stats.buffer_max);
-  option::Parser parse(usage, argc, argv, &options[0], &buffer[0]);
+  PonoOptions pono_options;
+  ProverResult res = pono_options.parse_and_set_options(argc, argv);
+  if (res == ERROR) return res;
+  assert(res == pono::UNKNOWN);
```

Comment:
Could you add one comment before this assert just clarifying that it should be unknown if the options are all set and error otherwise?